tests/reftests/rec-pin.test

```diff
@@ -0,0 +1,994 @@
+pin1sub1sub2 is now pinned to file://${BASEDIR}/pinnes (a/k) (version l1)
+Package pin1sub2 does not exist, create as a NEW package? [Y/n] y
+pin1sub2 is now pinned to git+file://${BASEDIR}/pinnes#master (b) (version l1)
+### opam list -s
+### opam pin
+pin1.l1              (uninstalled)  git    git+file://${BASEDIR}/pinnes#master
+pin1sub1.l1          (uninstalled)  rsync  file://${BASEDIR}/pinnes (a)
+pin1sub1sub1.l1      (uninstalled)  git    git+file://${BASEDIR}/pinnes#master (a/i)
+pin1sub1sub1sub1.l1  (uninstalled)  git    git+file://${BASEDIR}/pinnes#master (a/i/j)
+pin1sub1sub2.l1      (uninstalled)  rsync  file://${BASEDIR}/pinnes (a/k)
+pin1sub2.l1          (uninstalled)  git    git+file://${BASEDIR}/pinnes#master (b)
+### opam unpin -n --recursive pinnes
+Ok, pin1 is no longer pinned to git+file://${BASEDIR}/pinnes#master (version l1)
+Ok, pin1sub1 is no longer pinned to file://${BASEDIR}/pinnes (a) (version l1)
+Ok, pin1sub1sub1 is no longer pinned to git+file://${BASEDIR}/pinnes#master (a/i) (version l1)
+Ok, pin1sub1sub1sub1 is no longer pinned to git+file://${BASEDIR}/pinnes#master (a/i/j) (version l1)
+Ok, pin1sub1sub2 is no longer pinned to file://${BASEDIR}/pinnes (a/k) (version l1)
+Ok, pin1sub2 is no longer pinned to git+file://${BASEDIR}/pinnes#master (b) (version l1)
+### opam list -s
+### opam pin
+### opam pin  --recursive pinnes | unordered
+This will pin the following packages: pin1, pin1sub1, pin1sub1sub1, pin1sub1sub1sub1, pin1sub1sub2, pin1sub2. Continue? [Y/n] y
+Package pin1 does not exist, create as a NEW package? [Y/n] y
+pin1 is now pinned to git+file://${BASEDIR}/pinnes#master (version l1)
+Package pin1sub1 does not exist, create as a NEW package? [Y/n] y
+pin1sub1 is now pinned to file://${BASEDIR}/pinnes (a) (version l1)
+Package pin1sub1sub1 does not exist, create as a NEW package? [Y/n] y
+pin1sub1sub1 is now pinned to git+file://${BASEDIR}/pinnes#master (a/i) (version l1)
+Package pin1sub1sub1sub1 does not exist, create as a NEW package? [Y/n] y
+pin1sub1sub1sub1 is now pinned to git+file://${BASEDIR}/pinnes#master (a/i/j) (version l1)
+Package pin1sub1sub2 does not exist, create as a NEW package? [Y/n] y
+pin1sub1sub2 is now pinned to file://${BASEDIR}/pinnes (a/k) (version l1)
+Package pin1sub2 does not exist, create as a NEW package? [Y/n] y
+pin1sub2 is now pinned to git+file://${BASEDIR}/pinnes#master (b) (version l1)
+
+The following actions will be performed:
+  - install pin1sub1sub1     l1 (pinned)
+  - install pin1             l1 (pinned)
+  - install pin1sub1sub1sub1 l1 (pinned)
+  - install pin1sub1         l1 (pinned)
+  - install pin1sub2         l1 (pinned)
+  - install pin1sub1sub2     l1 (pinned)
+===== 6 to install =====
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> retrieved pin1.l1  (no changes)
+-> installed pin1.l1
+-> retrieved pin1sub1.l1  (file://${BASEDIR}/pinnes(a))
+-> installed pin1sub1.l1
```

Comment:
I’m not sure how it’s able to build this package given its build instructions were:
```
build: [ "false" ]
```

Reply:
it's the non locked files that contains a `build: [ "false" ]`, and we want to check that locked file are taken. Locked files version have prefixed `l` versions.